src/python/pants/backend/python/rules/python_test_runner.py

```diff
@@ -105,14 +107,23 @@ def calculate_timeout_seconds(
     timeout_default=pytest.options.timeout_default,
     timeout_maximum=pytest.options.timeout_maximum,
   )
+
+  global_options = options_bootstrapper.bootstrap_options.for_global_scope()
```

Comment:
So, I don't think that we should consume the OptionsBootstrapper directly like this.

Instead, hows about declaring an `optionable_rule` for `GlobalOptionsRegistrar` https://github.com/pantsbuild/pants/blob/efa597142f0b49db8ff10280d1a2bb78a482bbc1/src/python/pants/option/global_options.py#L117-L119, and installing it by default somewhere?

We should also consider renaming it.

Reply:
What's wrong with consuming OptionsBootstrapper?

Reply:
It is (as its name implies) a tool to help bootstrap Options, rather than a source of options itself. By accessing the OptionsBootstrapper, a rule can gain access to a bunch of stuff that it might not have declared dependencies on.

We don't have a "public API" convention for types exposed by rules yet, but the OptionsBootstrapper is one of the things that we would want to call private.

Reply:
Can we actually query an instance of `GlobalOptionsRegistrar` to get the value of an option? The type returned by `options.for_global_scope` is `OptionValueContainer`, but that type is used for things other than specifically global scope options; so I'm not sure if it makes sense to make that type a root rule. Also, we already have at least one rule that specifies `OptionsBootstrapper` as an input. I think if we want to disallow consuming OptionsBootstrapper, we should do it in another commit that replaces it everywhere in the codebase, and also creates some new type specifically to represent a collection of global options.

Reply:
I think it'd be great to be able to request a `GlobalOptionsRegistrar`. I didn't realize that was a possibility but agree with Stu that requesting `OptionsBootstrapper` is getting more things than we really need for here or the other `dist.py` rule.

> I think if we want to disallow consuming OptionsBootstrapper, we should do it in another commit that replaces it everywhere in the codebase

That'd be great if you have the time. It could either be a precursor or followup to this PR (precursor would be more sound).

Reply:
Well we can't request a `GlobalOptionsRegistrar` right now, we'd have to build a RootRule for it - but I don't think that's the type we want anyway for this sort of use case. I don't think we can query `GlobalOptionsRegistrar` to ask for the value of a global option. If we can I don't know what the API is for doing so.

Reply:
> Well we can't request a GlobalOptionsRegistrar right now, we'd have to build a RootRule for it

It would just be a `@rule` I think, since it can be computed from the `OptionsBootstrapper`. You might expose it from https://github.com/pantsbuild/pants/blob/master/src/python/pants/engine/legacy/options_parsing.py , which is where the rules that compute `optionable_rule`'s inputs live.

Reply:
@Eric-Arellano @stuhood updated this PR to get global options via a new type that can be the input to a rule